tests/integration/testlib/assertlib.py

```diff
@@ -0,0 +1,75 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+import collections
+import copy
+import six
+
+from . import statelib
+from .statelib import INTERFACES
+
+
+def assert_state(desired_state):
+    """Given a state, assert it against the current state."""
+    filtered_current_state = statelib.filtered_current_state(desired_state)
+    normalized_desired_state = _normalize_desired_state(desired_state,
+                                                        filtered_current_state)
+
+    assert normalized_desired_state == filtered_current_state
+
+
+def _normalize_desired_state(desired_state, current_state):
```

Comment:
Isn't something like this part of libnmstate?

Reply:
We want the test to assert the outcome even though the production code is already doing that (you pointed that out well in the last review).
This is the test version of such an assert which require normalization of the data.
Hopefully, we will not repeat bugs in production here, or at least have different bugs.

I also hope we will have less logic here, as we do not need the inner metadata and other special handling, we expect nmstate to return back something that is simpler.

Reply:
IMHO normalization does not describe this. For me it means that the state is changed to an equivalent state.

Reply:
Btw. is it expected that this will only work in the docker environment but not with real NICs? I guess then the ethernet subtree will be added and might break the expectations.

Reply:
I have renamed things in #111 , lets discuss it there.
I am not sure we have a problem with `ethernet`, from "outside" I am not sure the special handling is relevant (there are no None/s and stuff). But lets see how it works out through an added test later on.